m5paperwardriver/m5paperwardriver.ino

```diff
@@ -186,3 +214,28 @@ void loop() {
 
   delay(2000);
 }
+
+const char* getAuthType(uint8_t wifiAuth) {
+  switch (wifiAuth) {
+    case WIFI_AUTH_OPEN:
+      return "[OPEN]";
+    case WIFI_AUTH_WEP:
+      return "[WEP]";
+    case WIFI_AUTH_WPA_PSK:
+      return "[WPA_PSK]";
+    case WIFI_AUTH_WPA2_PSK:
+      return "[WPA2_PSK]";
+    case WIFI_AUTH_WPA_WPA2_PSK:
+      return "[WPA_WPA2_PSK]";
+    case WIFI_AUTH_WPA2_ENTERPRISE:
+      return "[WPA2_ENTERPRISE]";
+    case WIFI_AUTH_WPA3_PSK:
+      return "[WPA3_PSK]";
+    case WIFI_AUTH_WPA2_WPA3_PSK:
+      return "[WPA2_WPA3_PSK]";
+    case WIFI_AUTH_WAPI_PSK:
+      return "[WAPI_PSK]";
+    default:
+      return "[UNKNOWN]";
```

Comment:
this function is a great idea, but is it possible to return a string version of wifiAuth in the default case so we can manually figure it out and then more easily add support?

Reply:
In this statement `switch (wifiAuth) {` you read from a variable and pull a bunch of known cases out, which is great.  But in the case of default, you return only `[UNKNOWN]'.  it might be more useful to simply return the value from wifiAuth, or the value from wifiAuth with a known prefix or suffix indicating it's a raw value.  I don't know if you implemented all the current values from the source code, but even if you did it is missing a few which may be added later and returning the raw value would be more useful to the user than a generic "[UNKNOWN]"

Reply:
OK.

Reply:
> In this statement `switch (wifiAuth) {` you read from a variable and pull a bunch of known cases out, which is great. But in the case of default, you return only `[UNKNOWN]'. it might be more useful to simply return the value from wifiAuth, or the value from wifiAuth with a known prefix or suffix indicating it's a raw value. I don't know if you implemented all the current values from the source code, but even if you did it is missing a few which may be added later and returning the raw value would be more useful to the user than a generic "[UNKNOWN]"

1. They’re written in string format to the CSV. 

2. Yes, because these are what the esp32 WiFi library supports: 

- https://docs.espressif.com/projects/arduino-esp32/en/latest/api/wifi.html

- https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/network/esp_wifi.html

- https://www.arduino.cc/reference/en/libraries/wifi/wifi.scannetworks/

As for dwell times 200 is the standard, but in tests around 150 was about perfect for many M5 units. Tested this morning with no improvement. 

ChatGPT told me same as it told you “increasing the dwell time to 200ms provides a better opportunity to capture essential packets, as losing even 50% of the transmissions could still allow you to detect the access point​(
[Kismet](https://www.kismetwireless.net/docs/readme/intro/passive_capture/)
).”

To that end I’d recommend opening a PR on the linked wardriver code by Hewitt as it’s deployed worldwide. This is a single device hobby project. 

Regards.

Reply:
thanks for the suggestion, I opened an issue to discuss the dwell change.  I don't have a ton of experience with the embedded wifi stuff so I wanted feedback in case there was something different which I don't know.
https://github.com/JosephHewitt/wardriver_rev3/issues/177

Reply:
Nice work. It’s safer to dwell longer with more interfaces that’s for sure. I run these faster because loss has been negligible/nil. On larger rigs I notice a bunch of issues with interference when hopping faster then 5 ch/s.